ja_JP/readme.md

```diff
@@ -0,0 +1,738 @@
+`error`は、コンストラクタ、正規表現、エラーメッセージ、バリデーション関数などにも出来ます。
+
+### .doesNotThrow(function|promise, [message])
+
+`error`を投げない`関数`か`promise`のresolveであるとします。
+
+### .ifError(error, [message])
+
+`error`が偽と判断できる値とします。
+
+## アサーションのスキップ
+
+`skip`修飾子を使うことでどのようなアサーションもスキップできます。スキップされたアサーションはカウントされますので、計画したアサーションの数を変更する必要はありません。
+
+
+```js
+test(t => {
+  t.plan(2);
+  t.skip.is(foo(), 5); // スキップするときにplanの数を変更する必要はありません。
+  t.is(1, 1);
+});
+```
+
+## 強化されたassert
+
+AVAには、より叙述的なアサーションメッセージを与える、[`power-assert`](https://github.com/power-assert-js/power-assert)がビルトインとして付属してます。それは、テストを読んで、コードからより多くの情報を推論しようとします。
+
+以下のようなテストの場合:
+
+```js
+test(t => {
+	const x = 'foo';
+	t.ok(x === 'bar');
+});
+```
+
+通常通り手助けにならない出力が得られます:
+
+```
+false === true
+```
+
+強化されたassertでは以下のような結果が得られます:
+
+```
+t.ok(x === 'bar')
+     |
+     "foo"
+```
+
+このような場合には`t.is()`を使えて、またそうすべきでしょうが、これはただの簡単な例です。
+
+さらに高度な例を試しましょう:
+
+```js
+test(t => {
+	const a = /foo/;
+	const b = 'bar';
+	const c = 'baz';
+	t.ok(a.test(b) || b === c);
+});
+```
+
+その結果は以下です:
+
+```
+t.ok(a.test(b) || b === c)
+       |    |     |     |
+       |    "bar" "bar" "baz"
+       false
+```
+
+全てのassertのメソッドは強化されます。
+
+楽しんでください!
+
+
+## 隔離された環境
+
+それぞれのテストファイルは分離されたNode.jsのプロセスで実行されます。これには多くの利点があります。それぞれのテストファイルはお互いに影響し合うことがありません。グローバル環境でモック、ビルトインを上書きしたり等。ですが、このようになっているのは、主にパフォーマンスが理由です。Node.jsは非同期IOを同時に実行できますが、テストがメインスレッドをブロックするような、同期処理で重い場合は助けになりません。テストを同時に実行して並行でテストすることで、現代システムの利点を最大限に活用します。
+
+
+## Tips
+
+### 一時ファイル
+
+テストを並行で行うことは幾つかの課題が伴いますが、IOがその1つです。一般的に、直列のテストはテストのカレントディレクトリに一時的なディレクトリを作成して、終了時にそれを削除します。テストがお互いに競合するので並行でそれを行うことはできません。正しく行うには、それぞれが新しい一時的なディレクトリを利用することです。[`tempfile`](https://github.com/sindresorhus/tempfile)や[`temp-write`](https://github.com/sindresorhus/temp-write)といったモジュールが役立つでしょう。
+
+### デバッグ
+
+AVAはデフォルトで並列にテストを実行しますが、何かしらデバッグする必要があるときには次善策です。代わりに`--serial`オプションをつけてテストを直列で実行してください:
+
+```
+$ ava --serial
+```
+
+### コードカバレッジ
+
+AVAが[実行するテストファイル](#isolated-environment)のコードカバレッジに[`istanbul`](https://github.com/gotwarlost/istanbul)を使うことは出来ませんが、代わりに基本的にサブプロセスのサポートがある`istanbul`である[`nyc`](https://github.com/bcoe/nyc)を利用できます。
+
+バージョン`5.0.0`では、トランスパイルに関係なく、実際のコードのカバレッジをレポートするのにソースマップを使います。テスト対象のコードがインラインのソースマップかソースマップファイルへの参照のいずれかを含んでいることを確認してください。`babel/register`を使っていれば、`.babelrc`で`sourceMaps`オプションを`inline`に設定することができます。
+
+
+## FAQ
+
+### なぜ`mocha`、`tape`、`node-tap`などではないのか?
+
+Mochaはデフォルトのインターフェース(多くの人々が利用している)で`describe`や`it`のような暗黙のグローバルを利用することを必須としていて、主張が無さすぎて、肥大化していて、デフォルトで同期で、直列でテストを実行して、そして遅いです。Tapeやnode-tapはかなり良いです。AVAはそれらのシンタックスに強く影響を受けています。ですが、それらはいずれも、テストを直列で実行し、[TAP](https://testanything.org/)を第1級オブジェクトとして作り上げ、私の見方ですが、それがそのコードベースを複雑で結合したものにしました。TAPの出力は読みづらいので、外部のレポーターを利用することになります。AVAは自己主張が強く並行です。デフォルトの簡単なレポーターがあり、将来的にはTAPを通したレポーターをサポートします。
+
+### 名前はどのように書いて、どう発音するのか?
+
+AVAで、Avaやavaではありません。発音は[`/ˈeɪvə/` ay-və](https://github.com/sindresorhus/ava/blob/master/media/pronunciation.m4a?raw=true)です。
+
+### ヘッダーの背景は何?
+
+[アンドロメダ銀河です。](https://simple.wikipedia.org/wiki/Andromeda_galaxy)
```

Comment:
There is a link to Wikipedia in Japanese. it may be better? (`https://ja.wikipedia.org/wiki/%E3%82%A2%E3%83%B3%E3%83%89%E3%83%AD%E3%83%A1%E3%83%80%E9%8A%80%E6%B2%B3`)